components/roode/roode.cpp

```diff
@@ -31,9 +31,8 @@ namespace esphome
             {
                 // If the sensor could not be initialized print out the error code. -7 is timeout
                 ESP_LOGE(SETUP, "Could not initialize the sensor, error code: %d", sensor_status);
-                while (1)
-                {
-                }
+                this->mark_failed();
+                return;
```

Comment:
Did you know about this and avoided it for a reason? API states it logs the error and then update/loop/etc. functions are no longer called. This felt better than blocking the entire runtime.